src/MSCorEE.Desktop/MSCorEE.cs

```diff
@@ -72,5 +79,37 @@ public static partial class MSCorEE
         /// </returns>
         [DllImport(nameof(MSCorEE), PreserveSig = true)]
         public static extern unsafe int StrongNameFreeBuffer(byte* pbMemory);
+
+        [DllImport(nameof(MSCorEE), CharSet = CharSet.Unicode, PreserveSig = true)]
+        public static extern int CLRCreateInstance(
+            [In, MarshalAs(UnmanagedType.LPStruct)] Guid rclsid,
+            [In, MarshalAs(UnmanagedType.LPStruct)] Guid riid,
+            [Out, MarshalAs(UnmanagedType.Interface)] out object pMetaHost);
+
+        /// <summary>
+        /// Gets the version number of the common language runtime (CLR) that is associated with the specified process handle. This function has been deprecated in the .NET Framework version 4.
+        /// </summary>
+        /// <param name="hProcess">A handle to a process.</param><param name="buffer">A buffer that contains the version number string upon successful completion of the method.</param><param name="bufferSize">The length of the version buffer.</param><param name="bufferLength">A pointer to the length of the version number string.</param>
+        /// <returns>
+        /// HRESULT
+        /// </returns>
+        /// <remarks>
+        /// .NET Framework Versions: 4.5, 4, 3.5 SP1, 3.5, 3.0 SP1, 3.0, 2.0 SP1, 2.0
+        /// </remarks>
+        [DllImport(nameof(MSCorEE), CharSet = CharSet.Unicode)]
+        public static extern HResult GetVersionFromProcess([In] IntPtr hProcess, [MarshalAs(UnmanagedType.LPWStr), Out] StringBuilder buffer, [In] uint bufferSize, out uint bufferLength);
+
+        /// <summary>
+        /// Gets the common language runtime (CLR) version information of the specified file, using the specified buffer. This function has been deprecated in the .NET Framework 4.
+        /// </summary>
+        /// <param name="fileName">The path of the file to be examined.</param><param name="buffer">The buffer allocated for the version information that is returned.</param><param name="bufferSize">The size, in wide characters, of szBuffer.</param><param name="bufferLength">The size, in bytes, of the returned szBuffer.</param>
+        /// <returns>
+        /// HRESULT
+        /// </returns>
+        /// <remarks>
+        /// .NET Framework Versions: 4.5, 4, 3.5 SP1, 3.5, 3.0 SP1, 3.0, 2.0 SP1, 2.0, 1.1
+        /// </remarks>
+        [DllImport("mscoree.dll", CharSet = CharSet.Unicode)]
+        public static extern HResult GetFileVersion([MarshalAs(UnmanagedType.LPWStr), In] string fileName, [MarshalAs(UnmanagedType.LPWStr), Out] StringBuilder buffer, [In] uint bufferSize, out uint bufferLength);
```

Comment:
You're using `StringBuilder` where I'd probably have used `char*`. I think I'm ok either way. But I wanted to call it out in case @vbfox had any suggestions.